test.sh

```diff
@@ -58,22 +58,86 @@ dlTftpFiles() {
   done
 }
 
-prepareDeps() {
-  rm -rf ${WORKSPACE}/build-deps
-  mkdir -p ${WORKSPACE}/build-deps/${REPO_NAME} 
-  for i in ${REPOS[@]}; do
-     cd ${WORKSPACE}/build-deps
-     if [ "${i}" != "${REPO_NAME}" ]; then
-         git clone https://github.com/RackHD/${i}.git
-         cd ${i}
-         git checkout ${GIT_REFSPEC}
-         npm install --production
-     fi
-  done
+preparePackages() {
+  if [ -z ${MANIFEST_FILE_URL} ];
+  then
+      rm -rf ${WORKSPACE}/build-deps
+      if [ -z "${MULTI}" ]; then
+        mkdir -p ${WORKSPACE}/build-deps/${REPO_NAME}
+      else
+        mkdir -p ${WORKSPACE}/build-deps
+        mkdir -p ${WORKSPACE}/build
+      fi
+      for i in ${REPOS[@]}; do
+         cd ${WORKSPACE}/build-deps
+         if [[ "${REPO_NAME}" != *"${i}"* ]]; then
+             git clone https://github.com/RackHD/${i}.git
+             cd ${i}
+             git checkout ${GIT_REFSPEC}
+             npm install --production
+         fi
+      done
+
+  else
+      pushd ${WORKSPACE}
+      curl --user ${BINTRAY_USERNAME}:${BINTRAY_API_KEY} -L "${MANIFEST_FILE_URL}" -o rackhd-manifest
+      ./build-config/build-release-tools/HWIMO-BUILD ./build-config/build-release-tools/application/reprove.py \
+      --manifest rackhd-manifest \
+      --builddir ${WORKSPACE}/build-deps \
+      --jobs 8 \
+      --force \
+      checkout \
+      packagerefs
+
+      for i in ${REPOS[@]}; do
+          pushd ${WORKSPACE}/build-deps/${i}
+          npm install --production
+          popd
+      done
+      popd
+  fi
+}
+
+prepareDeps(){
+  preparePackages
   dlTftpFiles
   dlHttpFiles
+  apiPackageModify
 }
 
+apiPackageModify() {
+  if [[ "${REPO_NAME}" == *"on-http"* ]] && [[ "${REPO_NAME}" == *"rackhd"* ]]; then
+    pushd ${WORKSPACE}/on-http/extra
+    sed -i "s/.*git symbolic-ref.*/ continue/g" make-deb.sh
```

Comment:
seems you are replace the $BRANCH in on-http/extra/make-deb.sh, but can you please add comment in code ?